app/helpers/patient_query_helper.rb

```diff
@@ -281,18 +281,36 @@ def advanced_filter(patients, filters, tz_offset)
         patients = advanced_filter_relative_date(patients, :symptom_onset, filter, tz_diff, :date)
       when 'continous-exposure'
         patients = patients.where(continuous_exposure: filter[:value].present? ? true : [nil, false])
+      when 'close-contact-with-known-case'
+        case filter[:additionalFilterOption]
+        when 'Exact Match'
+          patients = if filter[:value].blank?
+                       patients.where(primary_telephone: [nil, ''])
+                     else
+                       patients.where('patients.contact_of_known_case_id = ?', filter[:value])
+                     end
+        when 'Contains'
+          patients = if filter[:value].blank?
+                       patients.where(primary_telephone: [nil, ''])
+                     else
+                       patients.where('patients.contact_of_known_case_id like ?', "%#{filter[:value]}%")
+                     end
+        end
       when 'telephone-number'
-        patients = if filter[:value].blank?
-                     patients.where(primary_telephone: [nil, ''])
-                   else
-                     patients.where('patients.primary_telephone like ?', Phonelib.parse(filter[:value], 'US').full_e164)
-                   end
-      when 'telephone-number-partial'
-        patients = if filter[:value].blank?
-                     patients.where(primary_telephone: [nil, ''])
-                   else
-                     patients.where('patients.primary_telephone like ?', "%#{filter[:value]}%")
-                   end
+        case filter[:additionalFilterOption]
```

Comment:
I think this change is good but I'm also thinking that it might mess with some users' saved filters in prod. Not sure if it's worth trying to come up with a migration to update any users' saved filters when we make these kind of changes or at least let the PO team know to let the users know that any previously saved telephone number filter they have is not going to work anymore after this update. It's one thing if the front end breaks when trying to parse an outdated saved filter, but I think whats worse would be users not knowing when certain records should be showing up when they should be.

Reply:
This is a good point. Not sure how to best handle this. @holmesie thoughts?

Reply:
Telephone number is one of the most common advanced filters, so we can’t just make this change and not also fix the saved filters in my opinion. I’d say either add a migration that adjusts saved filters in the database, or move this to a ticket on the backlog with that as a requirement. I agree overall it’s a good change, just not sure we want to make it part of this PR.

Reply:
@timwongj would this be hard to do? I think this would be a worthwhile change for consistency's sake if it's not too much effort

Reply:
I think it shouldn't be too difficult to do, the work involved would be going through all the saved export presets that include an advanced filter and updating the structure of it, would just need to be tested well

Reply:
Agree with @kylietmo, let's separate it out into a different ticket.

Reply:
Changes reverted here: https://github.com/SaraAlert/SaraAlert/pull/701/commits/53df6cd663ad6f1de906c9361cebe8bf15a0b0f6